src/Collapse/include/gudhi/Flag_complex_sparse_matrix.h

```diff
@@ -224,9 +217,17 @@ class Flag_complex_sparse_matrix {
     // Iterate over the non-zero columns
     for (typename Sparse_row_matrix::InnerIterator it(sparse_row_adjacency_matrix_, rw_u); it; ++it) {
       Row_index rw_v = it.index();
+
+      bool dominated_edge_not_found = false;
+      try {
+        edge_to_index_map_.at(std::minmax(row_to_vertex_[rw_u], row_to_vertex_[rw_v]));
+      } catch (const std::out_of_range& oor) {
```

Comment:
Uh, you have `find` or `count` if you want to test if an element is present. It is surprising that u_set_dominated_edges_ gets removed for being redundant with critical_edge_indicator_ and the diff does not mention critical_edge_indicator_ anywhere, but that's possible, I didn't read it.

Reply:
My modification was not working. We do not have the same information in both structures...
Rollback on df758d75